File: Basic/JS/1014.js
/**
 * @author: kurumi
 * @date: 2025-11-29
 * @description: 1014 福尔摩斯的约会
 */
const rl = require("readline").createInterface({input: process.stdin});
const iter = rl[Symbol.asyncIterator]();
const readline = async () => (await iter.next()).value;

const isAlpha = str => str >= 'A' && str <= 'Z' || str >= 'a' && str <= 'z';
const isDigit = str => str >= '0' && str <= '9';

void (async function() {
  const str1 = await readline();
  const str2 = await readline();
  const str3 = await readline();
  const str4 = await readline();
  const dayList = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
  let day = "", hour = "", minute = "";
  let position = 1;
  let ch1, ch2, ch3, ch4;
  for (let i = 0; i < str1.length && i < str2.length; i++) {
    ch1 = str1.charAt(i);
    ch2 = str2.charAt(i);
    if (ch1 == ch2) {
      if (position == 1 && ch1 >= 'A' && ch1 <= 'G') {
        day = dayList[ch1.charCodeAt(0) - 65];
        position = 2;
      } else if (position == 2) {
        if (isDigit(ch1)) {
          hour = '0' + ch1;
          break;
        } else if (ch1 >= 'A' && ch1 <= 'N') {
          hour = ch1.charCodeAt(0) - 55;
          break;
        }
      }
    }
  }
  for (let i = 0; i < str3.length && i < str4.length; i++) {
    ch3 = str3.charAt(i);
    ch4 = str4.charAt(i);
    if (ch3 == ch4 && isAlpha(ch3)) {
      minute = i >= 0 && i < 10 ? '0' + i : i.toString();
      break;
    }
  }
  process.stdout.write(`${day} ${hour}:${minute}`);
})();

File: Basic/Python/1014.py
'''
@Author: kurumi
@Date: 2025-11-29
@Description: 1014 福尔摩斯的约会
'''
from typing import Dict, List

def main():
    str1: str = input()
    str2: str = input()
    str3: str = input()
    str4: str = input()
    dayList: List[str] = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
    day, hour, minute = '', '', ''
    position: int = 1
    for i in range(0, min(len(str1), len(str2))):
        if str1[i] == str2[i]:
            if position == 1 and str1[i] >= 'A' and str1[i] <= 'G':
                day = dayList[ord(str1[i]) - ord('A')]
                position = 2
            elif position == 2:
                if str1[i].isdigit():
                    hour = '0' + str1[i]
                    break
                elif str1[i] >= 'A' and str1[i] <= 'N':
                    hour = ord(str1[i]) - ord('A') + 10
                    break
    for i in range(0, min(len(str3), len(str4))):
        if str3[i] == str4[i] and str3[i].isalpha():
            minute = '0' + str(i) if i >= 0 and i < 10 else str(i)
    print(f'{day} {hour}:{minute}')

if __name__ == '__main__':
    main()